### ISSUE TITLE
Neovim: error when closing the connection

### ISSUE THREAD

**Issue by przepompownia** [vim-vdebug/vdebug#453]
```
Request from non-main thread.
Requests from different threads should be wrapped with nvim.async_call(cb, ...)
  File "/home/user/.vim/pack/bundle/opt/vdebug/python3/vdebug/connection.py", line 199, in run
    print("Exception caught")
  File "/home/user/.local/lib/python3.8/site-packages/pynvim/plugin/script_host.py", line 186, in write
    self.redirect_handler(data)
  File "/home/user/.local/lib/python3.8/site-packages/pynvim/plugin/script_host.py", line 77, in <lambda>
    sys.stdout = RedirectStream(lambda data: nvim.out_write(data))
  File "/home/user/.local/lib/python3.8/site-packages/pynvim/api/nvim.py", line 396, in out_write
    return self.request('nvim_out_write', msg, **kwargs)
Exception in thread Thread-1:
Traceback (most recent call last):
  File "/home/user/.vim/pack/bundle/opt/vdebug/python3/vdebug/connection.py", line 183, in run
    self.__peek_for_exit()
  File "/home/user/.vim/pack/bundle/opt/vdebug/python3/vdebug/connection.py", line 209, in __peek_for_exit
    self.__check_exit(self.__message_q.get_nowait())
  File "/home/user/.vim/pack/bundle/opt/vdebug/python3/vdebug/connection.py", line 216, in __check_exit
    raise Exception("Exiting")
Exception: Exiting
During handling of the above exception, another exception occurred:
Traceback (most recent call last):
  File "/usr/lib/python3.8/threading.py", line 932, in _bootstrap_inner
    self.run()
  File "/home/user/.vim/pack/bundle/opt/vdebug/python3/vdebug/connection.py", line 199, in run
    print("Exception caught")
  File "/home/user/.local/lib/python3.8/site-packages/pynvim/plugin/script_host.py", line 186, in write
    self.redirect_handler(data)
  File "/home/user/.local/lib/python3.8/site-packages/pynvim/plugin/script_host.py", line 77, in <lambda>
    sys.stdout = RedirectStream(lambda data: nvim.out_write(data))
  File "/home/user/.local/lib/python3.8/site-packages/pynvim/api/nvim.py", line 396, in out_write
    return self.request('nvim_out_write', msg, **kwargs)
  File "/home/user/.local/lib/python3.8/site-packages/pynvim/api/nvim.py", line 178, in request
    raise NvimError("request from non-main thread")
pynvim.api.common.NvimError: request from non-main thread
Vdebug stopped waiting for a connection
Press ENTER or type command to continue
```

**Comment by meldafert** [comment]
This is a duplicate of #426 .

**Comment by BlackIkeEagle** [comment]
should be fixed with #479
